Servers/providers/azure-ad-sso.provider.ts

```
@@ -0,0 +1,864 @@
+
+      // Check token length (reasonable bounds for JWT)
+      if (token.length < 100 || token.length > 8192) {
+        throw new SSOError(
+          SSOErrorType.AUTHENTICATION_FAILED,
+          'Token length is outside acceptable range',
+          this.providerType
+        );
+      }
+
+      // Check for dangerous characters
+      if (/[\x00-\x1f\x7f<>{}[\]\\]/.test(token)) {
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)